Add additional delta options for iracing

ConnorMolz opened the pull request:
## Description
I added to the common HUD to LapTimeDelta the delta to the optimal lap and to the last lap which can be selected in the config (here is a hint that its only used in iRacing).

## Type of Change
- [x] New feature
- [x] Other (please specify): Improvement of existing Overlay

## Checklist
- [x] My code follows the project's coding style.
- [x] I have tested my changes thoroughly.
- [x] I have updated any necessary documentation.
- [x] My changes introduce no new warnings.

## Additional Notes
I need to do some additional test at the evening today, I will send a message when I am done with it.

ConnorMolz committed:
Map Optimal and Last lap deltas

ConnorMolz committed:
Implement different into LapDeltaOverlay

ConnorMolz committed:
Remove artefact from different idea

ConnorMolz committed:
Add name to the authors

RiddleTime commented on Race Element.HUD.Common/Overlays/Driving/LapDeltaBar/LapDeltaOverlay.cs:
```diff
@@ -106,16 +106,39 @@ public sealed override bool ShouldRender()
+        }
+        else
+        {
+            switch (_config.Delta.DeltaType)
```
```csharp
delta = _config.Delta.DeltaType switch
{
    LapTimeDeltaConfiguration.DeltaTypes.BestLap => (float)TimeSpan.FromMilliseconds(SimDataProvider.LocalCar.Timing.LapTimeDeltaBestMS).TotalSeconds,
    LapTimeDeltaConfiguration.DeltaTypes.LastLap => (float)TimeSpan.FromMilliseconds(SimDataProvider.LocalCar.Timing.LapTimeDeltaLastMs).TotalSeconds,
    LapTimeDeltaConfiguration.DeltaTypes.OptimalLap => (float)TimeSpan.FromMilliseconds(SimDataProvider.LocalCar.Timing.LapTimeDeltaOptimalMs).TotalSeconds,
    _ => 0
};
```

RiddleTime commented on Race Element.HUD.Common/Overlays/Driving/LapDeltaBar/LapDeltaOverlay.cs:
```diff
@@ -106,16 +106,39 @@ public sealed override bool ShouldRender()
+        
     }
 
     private float GetDelta()
```
Might as well return a c# TimeSpan for `GetDelta()`

RiddleTime commented on Race Element.HUD.Common/Overlays/Driving/LapDeltaBar/LapTimeDeltaConfiguration.cs:
```diff
@@ -60,4 +63,10 @@ public sealed class ColorsGrouping
     }
 
     public LapTimeDeltaConfiguration() => GenericConfiguration.AllowRescale = true;
+    public enum DeltaTypes
```
Is this compatible with RaceRoom?

ConnorMolz replied:
I sadly do not drive RaceRoom, but for what I saw does RaceRoom not support this feature

RiddleTime replied:
RaceRoom is Free to play!

RiddleTime reviewed (commented):
Added a couple of remarks, some code style tips and maybe a question about RaceRoom compatibility.

You can use the `HideForGame` attribute on a config grouping to hide it for multiple games.
Within the overlay there is an attribute available that tells you the game when started 

```csharp
[HideForGame(Game.AmericanTruckSimulator | Game.EuroTruckSimulator2)]
[ConfigGrouping("Pit Limiter", "Adjust the behavior of the pit limiter")]
public PitLimiterGrouping Pitlimiter { get; init; } = new();
public sealed class PitLimiterGrouping
{
    [ToolTip("Actives the pit limiter drawing")]
    public bool Enabled { get; init; } = true;
}
```
    
```csharp
Game upshiftSupportedGames = Game.RaceRoom | Game.iRacing;
if (upshiftSupportedGames.HasFlag(GameWhenStarted))
{.... }
  ```

ConnorMolz committed:
Hide Delta Type Option for RaceRoom (https://github.com/RiddleTime/Race-Element/pull/283#pullrequestreview-3542501049)

ConnorMolz committed:
Adjust switch statement

(https://github.com/RiddleTime/Race-Element/pull/283#discussion_r2590891901)

RiddleTime committed:
refactor common lap delta hud

- added a new config grouping for data that hides for raceroom
- use flag option for game enum
- removed unused Spectator option

RiddleTime commented:
I've pulled the PR and pushed some refactors, moved the delta type to a new config grouping so it hides for RaceRoom and shows for iRacing.

RiddleTime committed:
refactor common delta delta bar

refactor, removed unused members
